src/Integration.Vsix/SonarLintTagger/IssueTagger.cs

```diff
@@ -20,213 +20,71 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
-using EnvDTE;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Adornments;
 using Microsoft.VisualStudio.Text.Tagging;
-using Sonarlint;
-using SonarLint.VisualStudio.Integration.Vsix.Helpers;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
-    ///<summary>
-    /// Tracks SonarLint errors for a specific buffer.
-    ///</summary>
-    /// <remarks><para>The lifespan of this object is tied to the lifespan of the taggers on the view. On creation of the first tagger,
-    /// it starts tracking errors. On the disposal of the last tagger, it shuts down.</para>
+    /// <summary>
+    /// Flyweight wrapper that provides data and notifications about tags
+    /// in a single view to a single consumer
+    /// </summary>
+    /// <remarks>
+    /// See the README.md in this folder for more information
     /// </remarks>
     internal sealed class IssueTagger : ITagger<IErrorTag>, IDisposable
     {
-        private readonly DTE dte;
-        private readonly TaggerProvider provider;
-        private readonly ITextBuffer textBuffer;
-        private readonly IEnumerable<SonarLanguage> detectedLanguages;
+        private IssuesSnapshot issues;
```

Comment:
Most of the product changes are in this class. Essentially, most of the code from the old version of this class was moved to the new _TextBufferIssueTracker_ class, leaving this class as a very thin disposable tagger.